google-cloud-spanner/src/main/java/com/google/cloud/spanner/MultiplexedSessionDatabaseClient.java

```diff
@@ -92,6 +97,9 @@ void onError(SpannerException spannerException) {
         // synchronizing, as it does not really matter exactly which error is set.
         this.client.resourceNotFoundException.set((ResourceNotFoundException) spannerException);
       }
+      // Mark multiplexed sessions for RW as unimplemented and fall back to regular sessions if
+      // UNIMPLEMENTED is returned.
+      this.client.maybeMarkUnimplementedForRW(spannerException);
```

Comment:
I think that this will cause any UNIMPLEMENTED error to disable multiplexed sessions for read/write transactions, as the implementation of the method just checks for an `UNIMPLEMENTED` error code, and this `onError(..)` method is used for all types of transactions.

Reply:
Yes this is indeed a problem. Added error message verification to avoid this situation